### PR TITLE
Test administration controllers

### PR DESCRIPTION
**Pull request by thierrymarianne**
To use the migrated configuration, send requests with `migrate_configuration` header set to `true` (a browser extension like [ModHeader](https://chrome.google.com/webstore/detail/modheader/idgpnmonknjnojddfkpgkljpfnnfcklj) might help) or set an environment variable `migrate_configuration` set to `true` in command-line.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -74,6 +75,10 @@ script:
   - php bin/phpunit -c tests/
   - SYMFONY_DEPRECATIONS_HELPER=disabled php bin/phpunit -c tests/phpunit-admin.xml
   - composer phpunit-sf
+  - cd /var/www/prestashop.unit.test && cp app/config/parameters_test.yml.dist app/config/parameters_test.yml
+  - php app/console cache:clear -e test --no-warmup
+  - SHOP_HOSTNAME="localhost/prestashop.unit.test" php app/console doctrine:migrations:migrate -e test -n -q
```
Comment (mickaelandrieu) [inline .travis.yml]: why SHOP_HOSTNAME is redefined with the protocol ?

File: admin-dev/index.php [hunk 2]
```diff
@@ -75,8 +80,15 @@
     Debug::enable();
 }
 require_once __DIR__.'/../app/AppKernel.php';
-//require_once __DIR__.'/../app/AppCache.php';
-$kernel = new AppKernel(_PS_MODE_DEV_?'dev':'prod', _PS_MODE_DEV_);
+
+$configDir = __DIR__.'/../app/config';
+if (file_exists($configDir.'/parameters_test.yml')) {
```
Comment (mickaelandrieu) [inline admin-dev/index.php]: this means your application will always be in 'test' mode, is it intended ?

File: src/Core/Legacy/LegacyBootstrap.php [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+use PrestaShop\PrestaShop\Core\ContainerBuilder;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate;
+use Symfony\Component\Yaml\Yaml;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Exception\IOException;
+use PrestaShopBundle\Cache\LocalizationWarmer;
+use PhpEncryption;
+use Tools;
+
+class LegacyBootstrap
```
Comment (mickaelandrieu) [inline src/Core/Legacy/LegacyBootstrap.php]: except if I'm wrong, ``Core`` will be removed, how about ``PrestaShopBundle\Legacy` namespace ?

File: .travis.yml (hunk 1 above)
Comment (thierrymarianne) [inline .travis.yml]: In database, protocol should not be mentioned. See https://github.com/thierrymarianne/PrestaShop/pull/1/files#diff-f2bd4988502d2f57c5a2dbdb4f2f5021R17

File: admin-dev/index.php (hunk 2 above)
Comment (thierrymarianne) [inline admin-dev/index.php]: You're right, a new environment variable has to be introduced.

File: admin-dev/index.php [hunk 4]
```diff
@@ -82,8 +82,9 @@
 require_once __DIR__.'/../app/AppKernel.php';
 
 $configDir = __DIR__.'/../app/config';
-if (file_exists($configDir.'/parameters_test.yml')) {
-    $environment = 'test';
+
+if (getenv('KERNEL_ENVIRONMENT') !== false) {
+    $environment = getenv('KERNEL_ENVIRONMENT');
```
Comment (mickaelandrieu) [inline admin-dev/index.php]: so `$environment` equals `migration` and not `test` anymore ?

File: tests/travis-ci-apache-vhost [hunk 5]
```diff
@@ -7,6 +7,7 @@
   Alias /php5-fcgi /usr/lib/cgi-bin/php5-fcgi
   FastCgiExternalServer /usr/lib/cgi-bin/php5-fcgi -host 127.0.0.1:9000 -pass-header Authorization
   SetEnv migrate_configuration true
+  SetEnv KERNEL_ENVIRONMENT migration
```
Comment (mickaelandrieu) [inline tests/travis-ci-apache-vhost]: or "test" ?

File: travis-scripts/run-selenium [hunk 6]
```diff
@@ -1,9 +1,22 @@
...
 
-/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_10.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :10 -ac -screen 0 1600x1200x16
-cd /var/www/prestashop.unit.test/tests/Selenium
-npm install
-DISPLAY=:10 npm run selenium &> /dev/null &
+# Start Xvfb
+/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_11.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :11 -ac -screen 0 1600x1200x16
+
+# Download Chrome driver
+wget http://chromedriver.storage.googleapis.com/2.25/chromedriver_linux64.zip
```
Comment (mickaelandrieu) [inline travis-scripts/run-selenium]: any kind of "latest" stuff ? I'm pretty sure this instruction will cause some issues in months ^^

File: travis-scripts/run-selenium [hunk 7]
```diff
@@ -1,9 +1,22 @@
...
+mv chromedriver_linux64.zip /tmp
+cd /tmp && unzip /tmp/chromedriver_linux64.zip
+
+# Download Selenium server
+wget http://selenium-release.storage.googleapis.com/3.0/selenium-server-standalone-3.0.1.jar
+
+# Run Selenium server
+java -jar -Dwebdriver.chrome.driver=/tmp/chromedriver selenium-server-standalone-3.0.1.jar &
```
Comment (mickaelandrieu) [inline travis-scripts/run-selenium]: make me think .. any chance to use the same server for both PHP and node tests ?

**Comment by mickaelandrieu** [comment]
dat bo goss :)

**Pull request closed without merging**